test/integration/runner.spec.js

```diff
@@ -0,0 +1,146 @@
+import { expect } from 'chai'
+import _ from 'lodash'
+
+import { ensureResources, createOrderByOrderNumber } from './test-utils.js'
+import { run } from '../../src/runner.js'
+import getLogger from '../../src/utils/logger.js'
+import { getApiRoot } from '../../src/utils/client.js'
+
+describe('runner', () => {
+  const apiRoot = getApiRoot()
+  const logger = getLogger()
+  let productIds
+  let shippingMethodId
+  const DAY_IN_MS = 24 * 60 * 60 * 1000
+
+  before(async () => {
+    const resources = await ensureResources(apiRoot, logger)
+    productIds = resources.productIds
+    shippingMethodId = resources.shippingMethodId
+  })
+
+  it('should create a template order from the checkout order', async () => {
+    const orderNumber = new Date().getTime().toString()
+    const checkoutOrder = await createOrderByOrderNumber(
+      apiRoot,
+      logger,
+      productIds,
+      shippingMethodId,
+      orderNumber
+    )
+    await run()
+
+    const {
+      body: { results: templateOrders },
+    } = await apiRoot
+      .orders()
+      .get({
+        queryArgs: {
+          where: `custom(fields(checkoutOrderRef(id="${checkoutOrder.id}")))`,
+          expand: 'paymentInfo.payments[*]',
+        },
+      })
+      .execute()
+
+    expect(templateOrders).to.have.lengthOf(4)
+    assertCustomFieldsEqual(checkoutOrder, templateOrders)
+    assertLineItemsEqual(checkoutOrder, templateOrders)
+    assertTemplateOrderCustomFields(checkoutOrder, templateOrders)
+    assertPayments(checkoutOrder, templateOrders)
+    const { body: checkoutOrderUpdated } = await apiRoot
+      .orders()
+      .withId({ ID: checkoutOrder.id })
+      .get()
+      .execute()
+    expect(checkoutOrderUpdated.custom.fields.isSubscriptionProcessed).to.be
+      .true
+  })
+
+  function assertPayments(checkoutOrder, templateOrders) {
+    const checkoutOrderPayment = _.cloneDeep(
+      checkoutOrder.paymentInfo.payments[0].obj
+    )
+    delete checkoutOrderPayment.id
```

Comment:
is there another way to avoid deletes but assert all fields ? I could not reason much if those deletes are the right way to go ? what was the main idea behind that type of testing ?

Reply:
I don't know, this seems to be easiest way to compare if 2 objects are equal. I delete because IDs are obviously not equal, same applies for timestamps etc. Otherwise I would need to write every field I want to compare.